packages/m2m-gateway/test/integration/events/getAttributeEvents.test.ts

```diff
@@ -0,0 +1,86 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { m2mGatewayApi, m2mEventApi } from "pagopa-interop-api-clients";
+import { generateId } from "pagopa-interop-models";
+import {
+  eventService,
+  expectApiClientGetToHaveBeenCalledWith,
+  mockInteropBeClients,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import { getMockM2MAdminAppContext } from "../../mockUtils.js";
+
+describe("getAttributeEvents integration", () => {
+  const mockAttributeEvent1: m2mEventApi.AttributeM2MEvent = {
+    id: generateId(),
+    eventTimestamp: new Date().toJSON(),
+    eventType: "ATTRIBUTE_ADDED",
+    attributeId: generateId(),
+  };
+
+  const mockAttributeEvent2: m2mEventApi.AttributeM2MEvent = {
+    id: generateId(),
+    eventTimestamp: new Date().toJSON(),
+    eventType: "ATTRIBUTE_ADDED",
+    attributeId: generateId(),
+  };
+
+  const mockEventManagerResponse: m2mEventApi.AttributeM2MEvents = {
+    events: [mockAttributeEvent1, mockAttributeEvent2],
+  };
+
+  const mockGetAttributeM2MEvents = vi
+    .fn()
+    .mockResolvedValue(mockEventManagerResponse);
+
+  mockInteropBeClients.eventManagerClient = {
+    getAttributeM2MEvents: mockGetAttributeM2MEvents,
+  } as unknown as PagoPAInteropBeClients["eventManagerClient"];
+
+  beforeEach(() => {
+    mockGetAttributeM2MEvents.mockClear();
+  });
+
+  it("Should succeed and perform API clients calls", async () => {
```

Comment:
You could unify the two tests into a single one, with it.each on `[generateId(), undefined]` for lastEventId